src/component/requests.js

```diff
@@ -15,32 +17,73 @@ export default class Requests extends React.Component {
     });
   }
 
+  _onScroll() {
+    const scrollableDomNode = React.findDOMNode(this);
+    const fromIndex = Math.ceil(scrollableDomNode.scrollTop / requestElementHeight) - 15;
+    this.props.setFromIndex((fromIndex < 0) ? 0 : fromIndex);
+  }
+
+  componentDidMount() {
+    const scrollableDomNode = React.findDOMNode(this);
+    scrollableDomNode.addEventListener('scroll', this._onScroll.bind(this));
+  }
+
+  componentWillUnmount() {
+    const scrollableDomNode = React.findDOMNode(this);
+    scrollableDomNode.removeEventListener('scroll', this._onScroll);
+  }
+
+  componentDidUpdate() {
+    const previousFilter = this.filter;
+    this.filter = this.props.requestData.filter;
+    const requests = React.findDOMNode(this);
+
+    if(previousFilter !== this.filter) {
+      requests.scrollTop = 0;
+    }
+  }
+
   render() {
 
-    let {requests, filter} = this.props;
-    const {setActiveRequest, config} = this.props;
+    const {setActiveRequest, config, requestData} = this.props;
+    let amountOfRequests;
+
+    if(requestData.filter) {
+      amountOfRequests = requestData.requests.length;
+    } else {
+      amountOfRequests = requestData.totalCount;
+    }
+
 
-    requests = requests.map((request) => {
+    const requests = requestData.requests.map((request, index) => {
```

Comment:
Please rename `requests` to `requestComponents` or sth along the lines. "requests" is just over-used.